dj_hetmech_app/views.py

```diff
@@ -26,22 +26,41 @@ class QueryPairView(APIView):
     http_method_names = ['get']
 
     def polish_pathcounts(self, source_id, target_id, pathcounts_data):
-        """Polish pathcounts data to make source/target consistent with
-        user queries.
+        """This function polishes pathcounts_data. The polishment includes:
+        * Copy nested fields in 'metapath' and 'dgp' to upper level;
+        * Make source/target consistent with query parameters in the URL;
+        * Remove redundant fields;
+        * Sort pathcounts by certain fields.
         """
 
         for entry in pathcounts_data:
-            # If necessary, swap "source_degree" and "target_degree"
-            # values in each pathcount entry's "dgp".
+            # Copy the two key/value pairs in entry['metapath'] to entry:
+            entry['metapath_abbreviation'] = entry['metapath']['abbreviation']
+            entry['metapath_name'] = entry['metapath']['name']
+
+            # Copy all key/values in entry['dgp'] to entry:
+            for key, value in entry['dgp'].items():
```

Comment:
Should just `entry.pop('dgp')` and avoid having to delete later?

Could do?

```python
entry.update(entry.pop('dgp'))
```

Reply:
Interesting, I never used `update()` method in dict before.